### ISSUE TITLE
react-autosize-textarea dependency is unmaintained and outdated

### ISSUE THREAD

**Issue by sirreal** [WordPress/gutenberg#61568]
### Description

@gziolo and @t-hamano identified that that this library in unmaintained. It's currently causing issues with type incompatibilities (#61486) and other problems.

We could fork and maintain a version of the package or migrate to another solution.

See:
- https://github.com/WordPress/gutenberg/issues/48009#issuecomment-1735720752
- https://github.com/WordPress/gutenberg/pull/48215#issuecomment-1827645994

See conversation at https://github.com/WordPress/gutenberg/pull/61486#discussion_r1596375955

**Comment by Mamaduka** [comment]
Maybe we can invest some time and fix the `react-textarea-autosize` [Firefox issue](https://github.com/Andarist/react-textarea-autosize/issues/366).

Unfortunately, the `field-sizing` property is still [unsupported](https://caniuse.com/?search=field-sizing) in some browsers. Solving this with just CSS would've been nice. Read more: https://chriscoyier.net/2023/09/29/css-solves-auto-expanding-textareas-probably-eventually/.

**Comment by tyxla** [comment]
I've experimented with removing `react-autosize-textarea` in favor of our built-in `TextareaControl` with `field-sizing` in #64208. 

It's a bummer that it's blocked by `field-sizing`'s browser support, but hopefully, this will change soon.

**Comment by ZebulanStanphill** [comment]
See also: https://github.com/WordPress/gutenberg/issues/39619

**Comment by Mamaduka** [comment]
I think this is a duplicate of #39619.

@sirreal, let's close this issue and keep the old one. There's no need to keep duplicates around.

**Comment by sirreal** [comment]
Duplicate of https://github.com/WordPress/gutenberg/issues/39619.
